### PR TITLE
Quick fix for handling the map type in input data

### PR DESCRIPTION
**Pull request by alice2008**
If the input data has map type and mixed type for its values, i.e. mixed of string or numbers, the default schema when creating dataframe is string type. This temporary solution is when knowing the expected schema, casting the values to the expected type.

### CODE REVIEW

File: photon-client/src/integTest/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderIntegTest.scala [hunk 1]
```diff
@@ -35,6 +35,17 @@ class AvroDataReaderIntegTest extends SparkTestUtils {
 
   import AvroDataReaderIntegTest._
 
+  @Test
+  def testFeedAvro(): Unit = sparkTest("feedTest") {
```
Comment (ashelkovnykov) [inline photon-client/src/integTest/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderIntegTest.scala]: Should choose another name for this test. 'Feed' is specific to LinkedIn. Furthermore, what the test is actually testing is reading Maps with multiple types or non-String types. Should also add some documentation explaining this.

File: photon-client/src/integTest/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderIntegTest.scala [hunk 2]
```diff
@@ -35,6 +35,17 @@ class AvroDataReaderIntegTest extends SparkTestUtils {
...
+    val dr = new AvroDataReader()
+    val dataPath = new Path(INPUT_DIR, "feed")
+
+    val featureConfigMap = Map(
+      "shard1" -> FeatureShardConfiguration(Set("xgboost_click"), hasIntercept = true))
```
Comment (ashelkovnykov) [inline photon-client/src/integTest/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderIntegTest.scala]: Can fit on one line

File: photon-client/src/integTest/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderIntegTest.scala [hunk 3]
```diff
@@ -35,6 +35,17 @@ class AvroDataReaderIntegTest extends SparkTestUtils {
...
+    val (df, _) = dr.readMerged(dataPath.toString, featureConfigMap, 1)
```
Comment (ashelkovnykov) [inline photon-client/src/integTest/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderIntegTest.scala]: Please add a newline after this line

File: photon-client/src/integTest/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderIntegTest.scala (hunk 1 above)
Comment (alice2008) [reply photon-client/src/integTest/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderIntegTest.scala]: done.

File: photon-client/src/integTest/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderIntegTest.scala (hunk 2 above)
Comment (alice2008) [reply photon-client/src/integTest/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderIntegTest.scala]: fixed

File: photon-client/src/integTest/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderIntegTest.scala (hunk 3 above)
Comment (alice2008) [reply photon-client/src/integTest/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderIntegTest.scala]: fixed.

File: photon-client/src/test/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderTest.scala [hunk 4]
```diff
@@ -173,7 +173,7 @@ object AvroDataReaderTest {
   private val UnionFieldFloatDouble = "unionFieldFloatDouble"
   private val UnionFloatDoubleValue = 43.5
   private val UnionFieldIntLongFloatDouble = "unionFieldIntLongFloatDouble"
-  private val UnionIntLongFloatDoubleValue = 5
+  private val UnionIntLongFloatDoubleValue = 5.0D
```
Comment (ashelkovnykov) [inline photon-client/src/test/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderTest.scala]: What was the reason for changing these two values?

**Comment by alice2008** [comment]
updated the unit tests with 2 records

File: photon-client/src/test/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderTest.scala (hunk 4 above)
Comment (alice2008) [reply photon-client/src/test/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderTest.scala]:
```  private val avroSchema: Schema = SchemaBuilder
    .record("testAvroSchema")
    .namespace("com.linkedin.photon.ml.avro.data")
    .fields()
    .name(IntField).`type`().intType().noDefault()
    .name(StringField).`type`().stringType().noDefault()
    .name(BooleanField).`type`().booleanType().noDefault()
    .name(DoubleField).`type`().doubleType().noDefault()
    .name(FloatField).`type`().floatType().noDefault()
    .name(LongField).`type`().longType().noDefault()
    .name(MapField).`type`().map().values().stringType().noDefault()
    .name(UnionFieldIntString).`type`().unionOf().intType().and().stringType().endUnion().noDefault()
    .name(UnionFieldIntBoolean).`type`().unionOf().intType().and().booleanType().endUnion().noDefault()
    .name(UnionFieldIntLong).`type`().unionOf().intType().and().longType().endUnion().noDefault()
    .name(UnionFieldFloatDouble).`type`().unionOf().floatType().and().doubleType().and().nullType().endUnion().noDefault()
    .name(UnionFieldIntLongFloatDouble).`type`().unionOf().intType().and().longType().and().floatType().and().doubleType().and().nullType().endUnion().noDefault()
    .name(NullableIntField).`type`().unionOf().intType().and().nullType().endUnion().noDefault()
    .name(NullableBooleanField).`type`().unionOf().booleanType().and().nullType().endUnion().noDefault()
    .name(NullableDoubleField).`type`().unionOf().doubleType().and().nullType().endUnion().noDefault()
    .name(NullableFloatField).`type`().unionOf().floatType().and().nullType().endUnion().noDefault()
    .name(NullableLongField).`type`().unionOf().longType().and().nullType().endUnion().noDefault()
    .name(FeaturesField).`type`().array().items(nameAndTermSchema).noDefault()
    .endRecord()``` In above avro schema, the map expected a string value type; and `UnionIntLongFloatDoubleValue` expect a dominant numeric type which is Double.

**Pull request merged**